src/Sylius/Bundle/CoreBundle/Context/CustomerAndChannelBasedCartContext.php

```diff
@@ -55,7 +55,7 @@ public function getCart(): OrderInterface
             throw new CartNotFoundException('Sylius was not able to find the cart, as there is no logged in user.');
         }
 
-        $cart = $this->orderRepository->findLatestCartByChannelAndCustomer($channel, $customer);
+        $cart = $this->orderRepository->findLatestNotEmptyCartByChannelAndCustomer($channel, $customer);
```

Comment:
To be honest, I'm not 100% sure if this change will not be a little BC Break so I would like to mark this place for review 😃

Reply:
I don't have a strong opinion, but we could leave the previous implementation and check object emptiness. On the other hand, we are saving object hydration, and it seems to be a cleaner solution. So 👍 from my side